packages/warehouses/src/warehouseClients/DatabricksWarehouseClient.ts

```diff
@@ -189,51 +213,46 @@ export class DatabricksWarehouseClient implements WarehouseClient {
             table: string;
         }[],
     ) {
-        let pool: odbc.Pool;
-        let results: Result<SparkSchemaResult>[];
-        try {
-            pool = await odbc.pool(this.connectionString);
-        } catch (e) {
-            throw new WarehouseConnectionError(e.message);
-        }
+        const { session, close } = await this.getSession();
+        let results: SchemaResult[][];
+
         try {
             const promises = requests.map(async (request) => {
-                let connection: odbc.Connection;
+                let query: IOperation | null = null;
                 try {
-                    connection = await pool.connect();
-                } catch (e) {
-                    throw new WarehouseConnectionError(e.message);
-                }
-                try {
-                    const columns = (await connection.columns(
-                        // @ts-ignore
-                        'SPARK', // This is always SPARK
-                        request.schema,
-                        request.table,
-                        '',
-                    )) as Result<SparkSchemaResult>;
-                    return columns;
+                    query = await session.getColumns({
+                        catalogName: request.database,
+                        schemaName: request.schema,
+                        tableName: request.table,
+                    });
+
+                    const result = (await query.fetchAll()) as SchemaResult[];
+
+                    return result;
                 } catch (e) {
                     throw new WarehouseQueryError(e.message);
                 } finally {
-                    await connection.close();
+                    if (query) query.close();
                 }
             });
             results = await Promise.all(promises);
         } finally {
-            await pool.close();
+            await close();
         }
+
         return results.reduce<WarehouseCatalog>(
             (acc, result, index) => {
                 const columns = Object.fromEntries<DimensionType>(
                     result.map((col) => [
                         col.COLUMN_NAME,
-                        mapFieldType(col.TYPE_NAME),
+                        mapFieldType(col.DATA_TYPE),
                     ]),
                 );
                 const { schema, table } = requests[index];
+
                 acc.SPARK[schema] = acc.SPARK[schema] || {};
                 acc.SPARK[schema][table] = columns;
+
                 return acc;
             },
             { SPARK: {} } as WarehouseCatalog,
```

Comment:
We probably need to change this as well since is hardcoded for 'SPARK'

Reply:
good catch